corehq/apps/reports/tasks.py

```diff
@@ -170,17 +170,16 @@ def export_all_rows_task(ReportClass, report_state, recipient_list=None, subject
     file = report.excel_response
     report_class = report.__class__.__module__ + '.' + report.__class__.__name__
 
-    if report.domain is None:
-        # Some HQ-wide reports (e.g. accounting/smsbillables) will not have a domain associated with them
-        # This uses the user's first domain to store the file in the blobdb
-        report.domain = report.request.couch_user.get_domains()[0]
```

Comment:
The `report` instance isn't passed anywhere so `report.domain` wouldn't be used down the line. It seems the `report_storage_domain` is just used to authenticate the user has permission to view the file `report.excel_response` in the `export_report` view.

Reply:
oh yeah, this is inside a task, so it doesn't look like anything could happen with the report instance after this.  I was worried it might be something like:

```python
def a():
    report = get_report()
    b(report)
    return report.render()

def b():
    report.domain = "mydomain"
```
where even if `b` doesn't do anything else with `report`, something higher up the call stack might.  Mutating state is hard to reason about.  But yeah, it looks safe in this context.

Reply:
I think that also can't happen because the instance is created within the function [here](https://github.com/dimagi/commcare-hq/blob/36f9a52214534f18718697a9892b9f63529c37a9/corehq/apps/reports/tasks.py#L165) via 
```
    report = object.__new__(ReportClass)
    report.__setstate__(report_state)
```
where `report_state` was the original instance serialized and `__setstate__` creates an independent copy of the original instance.